### PR TITLE
[ONNX] Clean up patch functions

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83136

Changes:

- Move namespace handling from `_new_node` to `_graph_op` for clarity
- Always require the `aten` namespace when creating aten ops. Remove the `aten` argument supplied in `_aten_op` for clarity
- Rename the `_ATTR_PATTERN` global
- Improve types
- Update `_add_attribute` to raise ValueErrors

### CODE REVIEW

**Comment by justinchuby** [comment]
cc @BowenBao

File: torch/onnx/_patch_torch.py [hunk 1]
```diff
@@ -75,11 +76,23 @@ def const_if_tensor(arg):
...
+    if arg is None:
+        return arg
+    if isinstance(arg, _C.Value):
+        return arg
+    return _graph_op(g, "Constant", value_z=arg)
+
+
 # Generate an ONNX ATen op node.
-def _aten_op(g, operator: str, *args, overload_name: str = "", **kwargs):
-    kwargs["aten"] = True
```
Comment (justinchuby) [inline torch/onnx/_patch_torch.py:80]: Remove the need for sorting out "aten" from kwargs by specifying the op domain

Comment (BowenBao) [reply torch/onnx/_patch_torch.py:80]: cc @thiagocrepaldi for awareness. Shouldn't affect anything unless there are manual construction of "ATen" ops elsewhere.

**Review by BowenBao — APPROVED** [review]
LGTM

File: torch/onnx/_patch_torch.py (hunk 1 above)
Comment (thiagocrepaldi) [reply torch/onnx/_patch_torch.py:80]: Have you tested this with Caffe2 build? I remember about corner cases in which, sometimes `aten` was present in `kwargs` and other times it wasn't, but the new op had to be an `aten` one. That was one of the spots that dragged the merge of aten fallback back in the day

Comment (justinchuby) [reply torch/onnx/_patch_torch.py:80]: CI was green so I suppose so?

Comment (thiagocrepaldi) [reply torch/onnx/_patch_torch.py:80]: We cannot assume code coverage is 100%, especially when Caffe2 is involved. I believe we do have an issue and the code flow description below will try to show that.

During export, when `g.at()` was called before this change, `aten=True` would be added to `kwargs`. Next, `graph_op(..., **kwargs)` was called, which called `new_node(..., **kwargs)` in turn. Following, `_new_node(..., **kwargs)` looked for `aten` entry inside `kwargs.items()` to use it as a `bool` flag argument for `_add_attribute(..., aten=aten)` calls made for all others `kwargs` entries. Finally, `_add_attribute(..., aten)` evaluated `if aten and is_caffe2_aten_fallback():` to perform type validation and convert float attributes to int ones. 

After this PR, this expression inside `_add_attribute` will always evaluate to `False`, including scenarios that we don't support publicly (aka `_is_caffe2_fallback()`). In that case, error scenarios for caffe2, including unsupported tensor attributes or float attributes, are not captured by the converter. Effectively, `_patch_torch.py::174-183` is silently bypassed, defeating the purpose of the `aten` attribute.

**Pull request closed without merging**